### ISSUE TITLE
Netflix multi subs not working

### ISSUE THREAD

**Issue by BenedictTsang** [dannvix/NflxMultiSubs#20]
The extension doesn't work these days. I removed the extension and added agian but it still doesn't work.
![netflix](https://user-images.githubusercontent.com/41530678/43042047-d6d8fbec-8da4-11e8-8ce4-ed63f712f874.png)

**Comment by dannvix** [comment]
Works for me with the same episode. Please provide console logs.

![](https://user-images.githubusercontent.com/80781/43042904-66bfc8d4-8dba-11e8-88d3-5ef9f699bcb1.png)

**Comment by hiiwave** [comment]
The issue is confirmed at my side. Here is the console logs:

```
NflxMultiSubs> Intercepted: https://assets.nflxext.com/en_us/ffe/player/html/cadmium-playercore-6.0009.801.051.js content.min.js:1:5420
NflxMultiSubs> FYI: "Loading failed for the <script>" error is intented (for player code patching) content.min.js:1:5420
來源為「https://assets.nflxext.com/en_us/ffe/player/html/cadmium-playercore-6.0009.801.051.js」的 <script> 載入失敗。 browse:3
NflxMultiSubs> Patched: callback content.min.js:1:5420
NflxMultiSubs> Patched: hi-res hack content.min.js:1:5420
於 https://www.netflix.com/browse 呼叫 navigator.requestMediaKeySystemAccess()，卻不傳入包含 audioCapabilities 或 videoCapabilities 的 MediaKeySystemConfiguration 的方法已被棄用，即將停止支援。 none:7:95525
XML 解析錯誤: 不符合語法
位置: https://www.netflix.com/api/msl/NFCDFF-02-/cadmium/manifest
行號 1, 第 1 個字:
manifest:1:1
此網站使用與滾動列連結的定位效果，可能無法與同步平移效果一同使用，請參考 https://developer.mozilla.org/docs/Mozilla/Performance/ScrollLinkedEffects 當中的詳細資訊，並與我們一起討論相關的工具與功能！
```

Any idea?

**Comment by dannvix** [comment]
Thank you @hiiwave .  Based on the logs, seems you're using Firefox. It still works for me with exactly the same episode and Firefox (as shown in the screenshot).

Recently I found the some Netflix episodes likely have multiple URLs (?)... For the episode you tested, can you use this URL [https://www.netflix.com/watch/70302981](https://www.netflix.com/watch/70302981) and try again?

![](https://user-images.githubusercontent.com/80781/43086576-43391d90-8ed0-11e8-96a3-c2414007bd8f.png)

**Comment by hiiwave** [comment]
@dannvix Wow, incredibly I tried that url and it worked.

Therefore for some reason, netflix directs me to https://www.netflix.com/watch/70303287 when I browsed the menu and watched this video, where the issue exists and the error log suggested that manifest 70302981 was not played.
But the plugin could work via https://www.netflix.com/watch/70302981.

**Comment by hiiwave** [comment]
The following code snippet shows the reason. Without an equivalent url list, how are you gonna solve it?
```javascript
const playingManifest = (manifest.movieId.toString() === manifestInUrl);
      if (!playingManifest) {
        console.log(`Ignored: manifest ${manifest.movieId} not playing`);
        // Ignore but store it.
        this.manifestList.push(manifest);
        return;
}
```

**Comment by hiiwave** [comment]
I wonder whether this check is necessary. What will happen if we just simply remove these lines?

**Comment by BenedictTsang** [comment]
Here is the screen caption.

Here is the log:
Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
content.min.js:1 NflxMultiSubs> Intercepted:
https://assets.nflxext.com/en_us/ffe/player/html/cadmium-playercore-6.0009.801.011.js
content.min.js:1 NflxMultiSubs> FYI: "net::ERR_BLOCKED_BY_CLIENT" error is
intented (for player code patching)
content.min.js:1 NflxMultiSubs> Patched: callback
content.min.js:1 NflxMultiSubs> Patched: hi-res hack
content.min.js:1 NflxMultiSubs> Injected: nflxmultisubs.min.js
nflxmultisubs.min.js:8 NflxMultiSubs> Linked:
pjhnilfooknlkdonmjnleaomamfehkli
nflxmultisubs.min.js:8 NflxMultiSubs> Player: loaded=6.0009.801.011
nflxmultisubs.min.js:8 NflxMultiSubs> Manifest: 80058265
nflxmultisubs.min.js:8 NflxMultiSubs> Ignored: manifest 80058265 not playing
nflxmultisubs.min.js:8 NflxMultiSubs> Resize: 1920x1080 (92195.10000000082)
nflxmultisubs.min.js:8 NflxMultiSubs> Resize: 1920x947 (92886.39999999941)
nflxmultisubs.min.js:8 NflxMultiSubs> Resize: 1365x947 (93786.29999999976)

Thanks

2018-07-24 16:44 GMT+08:00 hiiwave <notifications@github.com>:

**Comment by dannvix** [comment]
@hiiwave Thank you for more details.  The mentioned manifest checking is to prevent interference with preloaded manifest (please refer to #5).

We'd need to rearrange the flow of handling manifests... will do it later.

**Comment by dannvix** [comment]
We've managed to check the manifest ID in an alternative approach, and published the latest version (`v1.6.6`) to [Chrome Web Store](https://chrome.google.com/webstore/detail/nflxmultisubs-netflix-mul/pjhnilfooknlkdonmjnleaomamfehkli) and [Firefox Add-ons](https://addons.mozilla.org/en-US/firefox/addon/nflxmultisubs/) to address this issue.

Enjoy! :balloon:
